lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1516,3 +1538,68 @@ def _get_disk_id(node_features: search_space.SetSpace[str]) -> str:
             return features.DiskStandardSSDLRS.get_disk_id()
         else:
             raise LisaException("No disk feature present.")
+
+    @staticmethod
+    def _get_data_disk_size_by_iops(
+        node_features: search_space.SetSpace[str], data_disk_iops: int
+    ) -> int:
+        if DiskType.DISK_EPHEMERAL in node_features:
+            raise LisaException(f"{DiskType.DISK_EPHEMERAL} only can be OS disk type.")
+        elif DiskType.DISK_PREMIUM in node_features:
+            return DataDisk.get_size_by_iops(DiskType.DISK_PREMIUM, data_disk_iops)
+        elif DiskType.DISK_STANDARD_HDD in node_features:
+            return DataDisk.get_size_by_iops(DiskType.DISK_STANDARD_HDD, data_disk_iops)
+        elif DiskType.DISK_STANDARD_SSD in node_features:
+            return DataDisk.get_size_by_iops(DiskType.DISK_STANDARD_SSD, data_disk_iops)
+        else:
+            raise LisaException("No data disk feature present.")
+
+
+class DataDisk:
+    # refer https://docs.microsoft.com/en-us/azure/virtual-machines/disks-types
+    IOPS_SIZE_PREMIUM: Dict[int, int] = {
+        120: 4,
+        240: 64,
+        500: 128,
+        1100: 256,
+        2300: 512,
+        5000: 1024,
+        7500: 2048,
+        16000: 8192,
+        18000: 16384,
+        20000: 32767,
+    }
+    IOPS_SIZE_STANDARD_HDD: Dict[int, int] = {
+        500: 32,
+        1300: 8192,
+        2000: 16384,
+    }
+    IOPS_SIZE_STANDARD_SDD: Dict[int, int] = {
+        500: 4,
+        2000: 8192,
+        4000: 16384,
+        6000: 32767,
+    }
+
+    @staticmethod
+    def get_min_size(iops_dict: Dict[int, int], data_disk_iops: int = 120) -> int:
```

Comment:
It's called in this class only, a private method is clearer. And put it under other public methods.

Reply:
fix